### PR TITLE
Fix conformance test failures for Google.Protobuf

### PR DESCRIPTION
**Pull request by ObsidianMinor**
It also fixes the exception message for unsupported output formats in Google.Protobuf.Conformance.

cc: @jtattermusch 

Fixes https://github.com/protocolbuffers/protobuf/issues/6865

### CODE REVIEW

**Comment by jtattermusch** [comment]
@anandolee @rafi-kamal  can you review from protobuf perspective?

File: csharp/src/Google.Protobuf/CodedInputStream.cs [hunk 1]
```diff
@@ -577,7 +577,32 @@ public uint ReadFixed32()
         /// </summary>
         public bool ReadBool()
         {
-            return ReadRawVarint32() != 0;
+            int result = 0; // can't do AND on bytes, it gets implicitly upcasted to int
```
Comment (jtattermusch) [inline csharp/src/Google.Protobuf/CodedInputStream.cs]: good catch but the fix seems overly complex, just `ReadRawVarint64() != 0` should be enough.

FTR that is also what Java does.
https://github.com/protocolbuffers/protobuf/blob/3cae867740e409d9b9fbc1224c35357a49ce0ce2/java/core/src/main/java/com/google/protobuf/CodedInputStream.java#L782

**Comment by jtattermusch** [comment]
@ObsidianMinor  I simplified the CodeInputStream changes and also fixed the bool wrapper case. Please doublecheck.

File: csharp/src/Google.Protobuf/CodedInputStream.cs [hunk 2]
```diff
@@ -845,7 +845,7 @@ public bool MaybeConsumeTag(uint tag)
 
         internal static bool? ReadBoolWrapper(CodedInputStream input)
         {
-            return ReadUInt32Wrapper(input) != 0;
+            return ReadUInt64Wrapper(input) != 0;
```
Comment (jtattermusch) [inline csharp/src/Google.Protobuf/CodedInputStream.cs]: Unrelated to this PR, I think there might be a bug if ReadUInt64Wrapper returns null - I'll try to check that and add a test in a separate PR.

**Comment by jtattermusch** [comment]
@rafi-kamal  can you take a look (and get a permission from @dankurka  to merge this fix).

File: csharp/src/Google.Protobuf/CodedInputStream.cs (hunk 1 above)
Comment (ObsidianMinor) [reply csharp/src/Google.Protobuf/CodedInputStream.cs]: Right, I thought about that. Though I also thought that maybe reading an actual varint would be excessive work. The case 99% of the time is that we have at least one byte left in the buffer and the varint is a 1 or 0. ReadRawVarint64 needs at least 10 bytes and does bit shifts and other things just to ultimately check if the whole value has at least one bit set.

File: csharp/src/Google.Protobuf/CodedInputStream.cs (hunk 2 above)
Comment (ObsidianMinor) [reply csharp/src/Google.Protobuf/CodedInputStream.cs]: Currently it never does so I don't think that'll be a problem.

**Comment by jtattermusch** [comment]
Fine to merge now?

**Comment by rafi-kamal** [comment]
Yes, please go ahead.

**Pull request merged**